### ISSUE TITLE
Constraint.update(offset:) won't update constant. feature/0.40.0

### ISSUE THREAD

**Issue by uny** [SnapKit/SnapKit#282]
### New Issue Checklist

* [x] I have looked at the [Documentation](http://snapkit.io/docs)
* [x] I have read the [F.A.Q.](http://snapkit.io/faq)

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform                | ios
 Platform Version        | 10.0
 SnapKit Version         | feature/0.40.0
 Integration Method      | carthage
 

### Issue Description
I want to update constraint constant.
[This line](https://github.com/uny/SnapKitDebug-282/blob/ac15a2409ff1eb21dc8c98dcc1cd83ee5b0788da/SnapKitDebug/ViewController.swift#L34) doesn't work properly.
I debugged for a while, but I cannot found the solution.
I inserted `updateConstraintsIfNeeded` and `layoutIfNeeded`, but nothing changed.

This is an issue for beta. Can you help me?

**Comment by robertjpayne** [comment]
@uny can you please post your constraint code?

**Comment by uny** [comment]
I uploaded a sample repository.
https://github.com/uny/SnapKitDebug-282/blob/master/SnapKitDebug/ViewController.swift#L34
Is this enough as the answer?

**Comment by uny** [comment]
Debug:

* Using LayoutConstraint directly ([`usingSnapKit `](https://github.com/uny/SnapKitDebug-282/blob/c95cce14bc55ae984ce7180324f561dfa3398161/SnapKitDebug/ViewController.swift#L13) is `false`), it works properly.
* `Constraint.layoutConstraints` become empty.
  1. Make [`Constraint.layoutConstraints`])(https://github.com/uny/SnapKitDebug-282/blob/c95cce14bc55ae984ce7180324f561dfa3398161/Carthage/Checkouts/SnapKit/Source/Constraint.swift#L49) public
  2. [After `makeConstraints`](https://github.com/uny/SnapKitDebug-282/blob/c95cce14bc55ae984ce7180324f561dfa3398161/SnapKitDebug/ViewController.swift#L29), it has the constraint.
  3. [In `didTapButton`](https://github.com/uny/SnapKitDebug-282/blob/c95cce14bc55ae984ce7180324f561dfa3398161/SnapKitDebug/ViewController.swift#L46), however, it becomes empty.

**Comment by uny** [comment]
`LayoutConstraint` is value type, so `layoutConstraints` should not be `weakObjects`. Isn't it?

**Comment by uny** [comment]
[`ConstraintMaker.makeConstraints`](https://github.com/SnapKit/SnapKit/blob/ae3012f884922d36f3021b85924df9a68c68c96b/Source/ConstraintMaker.swift#L172) and [`ConstraintMakerFinalizable.constraint`](https://github.com/SnapKit/SnapKit/blob/ae3012f884922d36f3021b85924df9a68c68c96b/Source/ConstraintMakerFinalizable.swift#L46) use different `Constraint` instances.

```
(lldb) po constraints
▿ 2 elements
  ▿ 0 : <Constraint: 0x6080001dfe00>
  ▿ 1 : <Constraint: 0x6080001dfef0>
```

```
(lldb) po self.buttonTopSnapKitConstraint
▿ Optional<Constraint>
  ▿ some : <Constraint: 0x6000001df950>
```

**Comment by robertjpayne** [comment]
Fixed in 0.40.0.beta4
